app/src/main/java/com/example/suberduberuber/Activities/DashboardActivity.java

```diff
@@ -98,7 +116,24 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
                 int itemId = menuItem.getItemId();
 
                 if (itemId == R.id.menu_home) {
-                    navController.navigate(R.id.action_to_dest_home_page);
+                    userRepository.getCurrentUser().get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
+                        @Override
+                        public void onComplete(@NonNull Task<DocumentSnapshot> task) {
+                            User user = task.getResult().toObject(User.class);
+                            if(user.getDriver()) {
+                                navController.navigate(R.id.action_to_dest_home_page);
+                            } else {
+                                requestRepository.getRidersCurrentRequest(user).get().addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                                    @Override
+                                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                                        if (task.getResult().getDocuments().size() > 0) {
+                                            navController.navigate(R.id.action_to_ridePending_page);
```

Comment:
I think an else should be added to navigate to the dest page. When you click home without a request its just a blank fragment right now